Retrained L1 Tau Trigger Integration

Duchstf opened the pull request:
#### PR description:

Hello, this PR updates the Tau NN training in the L1 trigger. The main changes include:


- Adding HW option for the tau as well as the new e/m seeding for tau jets.
- Updated Tau NN training with the lastest training samples.
- GT formatting for Tau Jets.


#### PR validation:

The improvements in performance is documented in the following talks:

- https://indico.cern.ch/event/1246470/
- https://indico.cern.ch/event/1271855/#202-performance-report-algorit

The new cmssw performance has been cross-checked by the L1 team and the results are included in the [2023 L1 Annual Review.](https://indico.cern.ch/event/1257943/contributions/5343040/attachments/2633523/4562436/AlgorithmsAndPerformance_2023_2.pdf) 

The original PR with related discussions/checks is here: https://github.com/cms-l1t-offline/cmssw/pull/1087

Duchstf committed:
Fixes after rebase.

Duchstf committed:
Merge L1 Tau Trigger changes

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-41492/35361

ERROR: Build errors found during clang-tidy run.
```
L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc:81:17: error: use of undeclared identifier 'TauNNTFCache' [clang-diagnostic-error]
std::unique_ptr<TauNNTFCache> L1NNTauProducer::initializeGlobalCache(const edm::ParameterSet& cfg) {
                ^
L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc:84:10: error: no viable conversion from returned value of type 'typename _MakeUniq<SessionCache>::__single_object' (aka 'unique_ptr<tensorflow::SessionCache>') to function return type 'int' [clang-diagnostic-error]
  return std::make_unique<tensorflow::SessionCache>(graphPath);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc:115:7: warning: the 'empty' method should be used to check for emptiness instead of 'size' [readability-container-size-empty]
--
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02783/el8_amd64_gcc11/external/gcc/11.2.1-f9b9dfdd886f71cd63f5538223d8f161/lib/gcc/x86_64-redhat-linux-gnu/11.2.1/../../../../include/c++/11.2.1/bits/unique_ptr.h:962:34: error: no matching constructor for initialization of 'TauNNId' [clang-diagnostic-error]
    { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
                                 ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~
L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc:75:22: note: in instantiation of function template specialization 'std::make_unique<TauNNId, const char *, const tensorflow::SessionCache *&, std::basic_string<char> &, int &>' requested here
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

aloeliger commented:
@Duchstf I have been searching both the L1 fork and CMSSW and cannot find any instance of a `TauNNTFCache` does this code exist in the integration branch somewhere that I have missed? Is this something that you have introduced?

qvyz commented:
> @Duchstf I have been searching both the L1 fork and CMSSW and cannot find any instance of a `TauNNTFCache` does this code exist in the integration branch somewhere that I have missed? Is this something that you have introduced?

Hi, I was running in this issue when trying to test the GT emulator:
The definition of TauNNTFCache is present in the snapshot branch 3  in the l1t offline repo.
(https://github.com/cms-l1t-offline/cmssw/blob/9fc02faa5eff9a7ad19bb9cd43e6c0cfb9f78867/L1Trigger/Phase2L1ParticleFlow/interface/TauNNId.h#L8-L11) maybe this file was forgotten

aloeliger commented:
Okay, this particular structure was actual removed, via agreement about 5 months ago in https://github.com/cms-sw/cmssw/pull/40333. It seems like this change was never backported, and this branch was never properly rebased onto a more recent branch with these changes, and adapted for them. 

@Duchstf @EmyrClement  You need to rebase these changes on to a more recent branch, and you will very likely need to fix some merge conflicts, retest the compilation of the emulator, and adapt it to the changes that have happened in the main fork.

something like:
```
cmsrel CMSSW_13_2_0_pre1
cd CMSSW_13_2_0_pre1/src/
cmsenv && git cms-init
git cms-rebase -u 
Duchstf:Tau_L1_2023_cmssw
```

at which point you will likely need to introduce some fixes, and make sure it all compiles again. **Please make sure this work is backed up in a backup branch in case anything goes wrong!**. Once it is properly adjusted, try pushing the changes back here.

Just one of the perils of trying to maintain a parallel git fork/branch I guess.

Duchstf commented:
@aloeliger Thank you so much for tracking down the error!!

EmyrClement committed:
code-format

Duchstf committed:
Merge pull request #4 from EmyrClement/Tau_L1_2023_cmssw_duc

Fixes for TauNNTFCache

epalencia commented:
Please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33014/summary.html
**COMMIT**: 71c40f3747f613189ac72105b188ea65a9ba9131
**CMSSW**: CMSSW_13_2_X_2023-06-06-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/41492/33014/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially removed 2 lines from the logs
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3219909
 - DQMHistoTests: Total failures: 6
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3219881
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_2_X_2023-06-06-1100+703c17/57375/triggerResults): no differences found

EmyrClement commented:
@Duchstf can you confirm if [this change](https://github.com/cms-l1t-offline/cmssw/pull/1087/commits/f30191306006dd9151ce4de6b3b97e2c7e25393d) is already included here or not?  We should make sure it is before proceeding.

Duchstf committed:
quick fix for eta and phi

Duchstf commented:
@EmyrClement I added it now! Sorry I didn't get the chance to make the change before the tests were ran :(

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-41492/35822

- This PR adds an extra 44KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-41492/35822/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-41492/35822/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

Duchstf committed:
code formatting

EmyrClement commented:
Just to note that the `.pb` files used when running the simulation workflow (which is no longer the default) are updated by [this PR to cms-data](https://github.com/cms-data/L1Trigger-Phase2L1ParticleFlow/pull/7).

perrotta commented on L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc:
```diff
@@ -149,9 +186,163 @@ void L1NNTauProducer::fillDescriptions(edm::ConfigurationDescriptions& descripti
+  float pNNVec[80];
+  for (unsigned i0 = 0; i0 < 80; i0++)
+    pNNVec[i0] = float(lNNVector[i0]);
+  L1TauEmu::etaphi_t eta = etaphi_t(seed.eta() * L1TauEmu::etaphi_base);
+  L1TauEmu::etaphi_t phi = etaphi_t(seed.phi() * L1TauEmu::etaphi_base);
```
```suggestion
```
These two aren't needed any more now, or am I wrong?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-41492/35823

- This PR adds an extra 44KB to repository

perrotta commented on L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc:
```diff
@@ -149,9 +186,163 @@ void L1NNTauProducer::fillDescriptions(edm::ConfigurationDescriptions& descripti
+        float deta = parts[i0].eta() - eta_1;
+        float dphi = parts[i0].phi() - phi_1;
+        float dr2 = deta * deta + dphi * dphi;
+        pz_tot = pz_tot + (parts[i0].pt()) * (1 - dr2 * 0.5);
+        py_tot = py_tot + (parts[i0].pt()) * dphi;  //sin(dphi ));
+        px_tot = px_tot + (parts[i0].pt()) * deta;  //sin(deta ));
```
```suggestion
```
also these do not seem to be used anywhere...
After a very quick glance this code still need quite some cleaning

Duchstf committed:
delete unncessary codes

Duchstf committed:
remove more unnecessary codes

Duchstf commented:
@perrotta Thanks for the quick looks!! I removed more codes that are unnecessary, let me know if you have more comments!! 

Duc.

perrotta commented:
> @perrotta Thanks for the quick looks!! I removed more codes that are unnecessary, let me know if you have more comments!!
> 
> Duc.

Thank you Duc.
Not now, I'd let the usual reviewer to comment it first, release managers enter later in the game normally. I only went through it by chance and I noticed that some cleaning should have been done.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-41492/35826

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #41492 was updated. @epalencia, @AdrianoDee, @srimanob, @cmsbuild, @aloeliger can you please check and sign again.

aloeliger commented:
please test

cmsbuild commented:
-1

**Failed Tests**:   Build  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33026/summary.html
**COMMIT**: 1800c936b140319dae48d215d0ede23c5c95cd2e
**CMSSW**: CMSSW_13_2_X_2023-06-06-2300/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/41492/33026/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation warning when building:  See details on the summary page.

## Clang Build

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

aloeliger commented:
@Duchstf: The compiler finds unused variables:

```
L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-06-2300/src/L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc:209:9: warning: unused variable 'eta_1' [-Wunused-variable]
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-06-2300/src/L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc:210:9: warning: unused variable 'phi_1' [-Wunused-variable]
```

These need to be removed.

Duchstf committed:
remove eta and phi

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-41492/35827

- This PR adds an extra 28KB to repository

aloeliger commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33028/summary.html
**COMMIT**: 8debb0ef4fe2e4a72e04432e6716c6f05c5abb25
**CMSSW**: CMSSW_13_2_X_2023-06-06-2300/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/41492/33028/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially removed 8 lines from the logs
 - Reco comparison results: 7 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3219909
 - DQMHistoTests: Total failures: 6
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3219881
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_2_X_2023-06-06-2300+703c17/57381/triggerResults): no differences found

aloeliger commented:
This PR is (somewhat) dependent on https://github.com/cms-data/L1Trigger-Phase2L1ParticleFlow/pull/7

aloeliger commented:
Please test with https://github.com/cms-data/L1Trigger-Phase2L1ParticleFlow/pull/7

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33033/summary.html
**COMMIT**: 8debb0ef4fe2e4a72e04432e6716c6f05c5abb25
**CMSSW**: CMSSW_13_2_X_2023-06-07-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/41492/33033/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @gartung cms-sw/cmssw#41861

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33033/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33033/git-merge-result


## Comparison Summary



**Summary**:
 - You potentially removed 5 lines from the logs
 - Reco comparison results: 6 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3219909
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3219884
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_2_X_2023-06-07-1100+703c17/57388/triggerResults): no differences found

aloeliger commented:
+l1

This PR has been reviewed by L1 as a part of https://github.com/cms-l1t-offline/cmssw/pull/1087, and is necessary for the ongoing GT emulator PR https://github.com/cms-sw/cmssw/pull/41808 . At this point I think we would like to hear from others.

aloeliger commented:
@Duchstf Have you had a chance to look at upgrade's review comments?

Duchstf commented:
> @Duchstf Have you had a chance to look at upgrade's review comments?

Hi @aloeliger, I'm looking at it now. However I'm wondering if you could tell me what's the correct recipe to set up my branch with the main CMSSW is, I was using your suggestions earlier:

```
cmsrel CMSSW_13_2_0_pre1
cd CMSSW_13_2_0_pre1/src/
cmsenv && git cms-init
git cms-rebase-topic-u Duchstf:Tau_L1_2023_cmssw
```

However it gives me this errors:

```
(base) bash-4.2$ git cms-rebase-topic -u Duchstf:Tau_L1_2023_cmssw
From .
 * [new ref]                            -> rebase-attempt
From https://github.com/Duchstf/cmssw
 * [new branch]              Tau_L1_2023_cmssw -> Duchstf/Tau_L1_2023_cmssw
Switched to branch 'rebase-attempt'
Switched to a new branch 'Tau_L1_2023_cmssw'
Created branch Tau_L1_2023_cmssw to follow Tau_L1_2023_cmssw from repository Duchstf
Auto-merging DataFormats/L1TParticleFlow/interface/gt_datatypes.h
Auto-merging DataFormats/L1TParticleFlow/src/classes_def.xml
CONFLICT (content): Merge conflict in DataFormats/L1TParticleFlow/src/classes_def.xml
error: could not apply bd46e3501b7... Merge L1 Tau Trigger changes
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply bd46e3501b7... Merge L1 Tau Trigger changes
```

I'm pretty confused because the codes passed all the checks, so I don't really know where the problem is.

AdrianoDee commented:
Hi @Duchstf this is probably due to the fact that you are using `CMSSW_13_2_0_pre1` that is not that new. I'd suggest you to use one of the latest IBs to be better aligned to the latest developments (such as `MSSW_13_2_X_2023-06-20-1100`). You can always check which is the latest with something like `scram list | grep 13_2_X`.

aloeliger commented:
@Duchstf It is probably simpler to just check out the branch you already have, instead of rebasing it again?

```
cmsrel CMSSW_13_2_0_pre1
cd CMSSW_13_2_0_pre1/src/
cmsenv && git cms-init
git cms-checkout-topic-u Duchstf:Tau_L1_2023_cmssw
```

I tend to forget about the `cms-checkout-topic` command, but it is what you're looking for here I think.

Duchstf committed:
add comments and const

Duchstf commented:
@AdrianoDee I think I addressed all the comments you have, let me know if I need anything else!!

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-41492/36006

- This PR adds an extra 48KB to repository

cmsbuild commented:
Pull request #41492 was updated. @epalencia, @AdrianoDee, @srimanob, @cmsbuild, @aloeliger can you please check and sign again.

aloeliger commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT AddOn 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/summary.html
**COMMIT**: 70ee234268495241732b0b757c29d7967ac6f895
**CMSSW**: CMSSW_13_2_X_2023-06-20-2300/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/41492/33290/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals

- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/25.0_TTbar)
```
----- Begin Fatal Exception 21-Jun-2023 12:56:42 CEST-----------------------
An exception of category 'PluginNotFound' occurred while
   [0] Constructing the EventProcessor
Exception Message:
Unable to find plugin 'Pythia8ConcurrentGeneratorFilter' in category 'CMS EDM Framework Module'. Please check spelling of name.
----- End Fatal Exception -------------------------------------------------
```
- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/135.4_ZEEFS_13)
```
----- Begin Fatal Exception 21-Jun-2023 12:56:57 CEST-----------------------
An exception of category 'PluginNotFound' occurred while
   [0] Constructing the EventProcessor
Exception Message:
Unable to find plugin 'Pythia8ConcurrentGeneratorFilter' in category 'CMS EDM Framework Module'. Please check spelling of name.
----- End Fatal Exception -------------------------------------------------
```
- [5.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/5.1_TTbarFS)
```
----- Begin Fatal Exception 21-Jun-2023 12:56:57 CEST-----------------------
An exception of category 'PluginNotFound' occurred while
   [0] Constructing the EventProcessor
Exception Message:
Unable to find plugin 'Pythia8ConcurrentGeneratorFilter' in category 'CMS EDM Framework Module'. Please check spelling of name.
----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more relval errors ...</summary>

- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/1330.0_ZMM_13)
- [2018.1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/2018.1_TTbarFS_13_UP18)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/25202.0_TTbar_13)
- [10024.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/10024.0_TTbar_13+2017)
- [10042.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/10042.0_ZMM_13+2017)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/10224.0_TTbar_13+2017PU)
- [10824.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/10824.0_TTbar_13+2018)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/11634.0_TTbar_14TeV+2021)
- [11634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/11634.911_TTbar_14TeV+2021_DD4hep)
- [11634.914](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/11634.914_TTbar_14TeV+2021_DDDDB)
- [11834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/11834.0_TTbar_14TeV+2021PU)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [12434.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/12434.7_TTbar_14TeV+2023_trackingMkFit)
- [13234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/13234.0_TTbar_14TeV+2021FS)
- [13434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/13434.0_TTbar_14TeV+2021FSPU)
- [23234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/23234.0_TTbar_14TeV+2026D94)
- [23634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/23634.0_TTbar_14TeV+2026D95)
- [23634.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/23634.911_TTbar_14TeV+2026D95_DD4hep)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/250202.181_TTbar13TeVPUppmx2018)
- [23834.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/23834.999_TTbar_14TeV+2026D95PU_PMXS1S2PR)
- [158.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/158.01_HydjetQB12ppRECOin2018reMINIAOD)
- [312.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2022)
- [1000.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrix-results/1000.0_RunMinBias2011A)
</details>



## RelVals-INPUT

- [140.106](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/140.106_RunJetMET2022D)```140.106_RunJetMET2022D/step2_RunJetMET2022D.log```
- [140.114](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/140.114_RunHcalNZS2022D)```140.114_RunHcalNZS2022D/step2_RunHcalNZS2022D.log```
- [140.116](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/140.116_RunCommissioning2022D)```140.116_RunCommissioning2022D/step2_RunCommissioning2022D.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [140.51](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/140.51_RunHI2010)
- [140.52](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/140.52_RunHI2010)
- [140.54](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/140.54_RunPA2013)
- [141.009](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/141.009_RunCosmics2023B)
- [10042.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10042.0_ZMM_13+2017)
- [10059.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10059.0_QCD_Pt_3000_3500_13+2017)
- [10025.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10025.0_ZEE_13+2017)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10224.0_TTbar_13+2017PU)
- [10026.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10026.0_QCD_Pt_600_800_13+2017)
- [10224.15](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10224.15_TTbar_13+2017PU_JMENano)
- [10225.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10225.0_ZEE_13+2017PU)
- [10803.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10803.0_SingleElectronPt1000+2018)
- [10804.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10804.0_SingleGammaPt10+2018)
- [10802.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/runTheMatrixINPUT-results/10802.0_SingleElectronPt35+2018)
</details>



## AddOn Tests

- [fastsim](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/fastsim)
```
----- Begin Fatal Exception 21-Jun-2023 12:56:12 CEST-----------------------
An exception of category 'PluginNotFound' occurred while
   [0] Constructing the EventProcessor
Exception Message:
Unable to find plugin 'Pythia8ConcurrentGeneratorFilter' in category 'CMS EDM Framework Module'. Please check spelling of name.
----- End Fatal Exception -------------------------------------------------
```
- [fastsim1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/fastsim1)
```
----- Begin Fatal Exception 21-Jun-2023 12:56:12 CEST-----------------------
An exception of category 'PluginNotFound' occurred while
   [0] Constructing the EventProcessor
Exception Message:
Unable to find plugin 'Pythia8ConcurrentGeneratorFilter' in category 'CMS EDM Framework Module'. Please check spelling of name.
----- End Fatal Exception -------------------------------------------------
```
- [fastsim2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/fastsim2)
```
----- Begin Fatal Exception 21-Jun-2023 12:56:12 CEST-----------------------
An exception of category 'PluginNotFound' occurred while
   [0] Constructing the EventProcessor
Exception Message:
Unable to find plugin 'Pythia8ConcurrentGeneratorFilter' in category 'CMS EDM Framework Module'. Please check spelling of name.
----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more addon errors ...</summary>

- [hlt_mc_Fake](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_Fake)
- [hlt_mc_Fake](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_Fake)
- [hlt_mc_Fake](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_Fake)
- [hlt_mc_Fake1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_Fake1)
- [hlt_mc_Fake1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_Fake1)
- [hlt_mc_Fake1](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_Fake1)
- [hlt_mc_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_Fake2)
- [hlt_mc_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_Fake2)
- [hlt_mc_Fake2](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_Fake2)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_GRun)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_GRun)
- [hlt_mc_GRun](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_GRun)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_HIon)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_HIon)
- [hlt_mc_HIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_HIon)
- [hlt_mc_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_PIon)
- [hlt_mc_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_PIon)
- [hlt_mc_PIon](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_PIon)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_PRef)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_PRef)
- [hlt_mc_PRef](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33290/addOnTests/hlt_mc_PRef)
</details>

srimanob commented:
@cmsbuild please test

It seems to be a glitch, let's try again.

srimanob commented:
CMSSW_13_2_X_2023-06-20-2300 is broken, we should wait until the IB comes.

aloeliger commented:
> CMSSW_13_2_X_2023-06-20-2300 is broken, we should wait until the IB comes.

I see [CMSSW_13_2_X_2023-06-21-1100](https://github.com/cms-sw/cmssw/releases/tag/CMSSW_13_2_X_2023-06-21-1100) and [CMSSW_13_2_X_2023-06-21-1600](https://github.com/cms-sw/cmssw/releases/tag/CMSSW_13_2_X_2023-06-21-1600) in the tags. Out of curiosity, when does buildbot pick up a new tag for IB's?

srimanob commented:
@cmsbuild please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-703c17/33310/summary.html
**COMMIT**: 70ee234268495241732b0b757c29d7967ac6f895
**CMSSW**: CMSSW_13_2_X_2023-06-21-1600/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/41492/33310/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



There are some workflows for which there are errors in the baseline:
 25034.114 step 2
25034.21 step 2
25034.9921 step 2
25034.999 step 2
25034.99 step 2
25061.97 step 1
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3200270
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3200245
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 207 log files, 159 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_2_X_2023-06-21-1600+703c17/57615/triggerResults): no differences found

AdrianoDee commented:
+upgrade

aloeliger commented:
+l1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @rappoccio (and backports should be raised in the release meeting by the corresponding L2)

rappoccio commented:
Can someone explain why hundreds of magic numbers are added to a dataformats class? This looks extremely suboptimal. At the very least it should be configurable, but in the best case scenario this should come from the CondDB.

srimanob commented:
I would propose to open git issue to follow up on performance and code cleaning. I agree that some part can migrate to db or cms data. However, this PR does not introduce new data file, but updates the one which are in the release already.

aloeliger commented:
@rappoccio What you are probably referring to are the neural network weights for the phase 2 taus? This phase 2 project was started before L1 settled on a definite HLS4ML emulation strategy developed for CICADA and so I think just tried to insert the firmware model into CMSSW as-is, so updating a training relies on changing these weights this way.

@EmyrClement can comment norepinephrine, but I believe that it is on the to-do list to migrate emulation strategies eventually.

rappoccio commented:
+1
